### PR TITLE
Rewrite main navigation

### PR DESCRIPTION
**Pull request by mattbowen-usds**
- Make the site navigation editable by non-programmers
- Make the "current" nav link highlighting work based on URL

### CODE REVIEW

File: tech-far-hub/src/components/navigation.tsx [hunk 1]
```diff
@@ -51,124 +58,34 @@ const Navigation = ({ isNavExpanded, onNavExpanded }: iNavigation) => {
...
-        }}
-        menuId="resourcesDropdown"
-        isOpen={isOpen[5]}
-        label="Resources"
-        isCurrent={false}
-      />
-      <Menu key="resources" items={resourcesSubItems} isOpen={isOpen[5]} />
-    </>,
-  ];
+  const yamlNavItems = navYAML.map((element, idx) => {
```
Comment (vim-usds) [inline tech-far-hub/src/components/navigation.tsx:61]: Wow! This YML approach cuts down a ton of code!

File: tech-far-hub/src/components/resouces-card.tsx [hunk 2]
```diff
@@ -12,9 +13,13 @@ interface IResourceNode {
...
 
 const ResouceCard = ({ node, width = 3 }: { node: IResourceNode; width?: ColumnSizes }): JSX.Element => {
-  if (node.html && node.frontmatter && node.frontmatter.href) {
+  if (node.html && node.frontmatter && node.frontmatter.href && node.frontmatter.heading) {
     return (
-      <Card headerFirst gridLayout={{ tablet: { col: width } }}>
+      <Card
+        headerFirst
+        gridLayout={{ tablet: { col: width } }}
+        key={`resource-${_.snakeCase(node.frontmatter.heading)}`}
```
Comment (vim-usds) [inline tech-far-hub/src/components/resouces-card.tsx:21]: Good ol' lodash, forgot about this lib!

File: tech-far-hub/src/components/resources.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ const Resources = () => {
   return (
     <CardGroup>
       {data.allMarkdownRemark.edges.map(({ node }) => {
-        return <ResouceCard node={node}></ResouceCard>;
+        return <ResouceCard node={node} key={node.id}></ResouceCard>;
```
Comment (vim-usds) [inline tech-far-hub/src/components/resources.tsx:38]: Typo with `<ResourceCard>` ?

File: tech-far-hub/src/types.ts [hunk 4]
```diff
@@ -51,10 +53,10 @@ export type MinimalPageQuery = {
   readonly siblings: { readonly nodes: ReadonlyArray<{ readonly frontmatter: IMinimalFrontmatter | null }> };
 };
 
-export type TOCEnhancedQueryPageProps<T extends MinimalPageQuery> = PageProps<
-  Omit<T, "currentPage"> & (DeepPick<T, "currentPage.!tableOfContents"> & TableOfContents),
```
Comment (vim-usds) [inline tech-far-hub/src/types.ts:55]: Did the original `DeepPick<T, "currentPage.!tableOfContents">` give a TS warning?

**Review by vim-usds — APPROVED** [review]
Killing it!

File: tech-far-hub/src/components/resources.tsx (hunk 3 above)
Comment (mattbowen-usds) [reply tech-far-hub/src/components/resources.tsx:38]: lol everywhere oops!

File: tech-far-hub/src/types.ts (hunk 4 above)
Comment (mattbowen-usds) [reply tech-far-hub/src/types.ts:55]: It did, and this does too. But now at least it's less hard to read.

File: tech-far-hub/src/components/resouces-card.tsx (hunk 2 above)
Comment (mattbowen-usds) [reply tech-far-hub/src/components/resouces-card.tsx:21]: They use it in some docs for gatsby and i was like "Oh right i love that library!"

**Pull request merged**